src/newViews/AnalysisModule/Analysis/AnalysisCloneModal/index.tsx

```diff
@@ -0,0 +1,164 @@
+    startDate: string;
+    endDate: string;
+};
+
+type FormSchema = ObjectSchema<PartialForm<FormType>>;
+type FormSchemaFields = ReturnType<FormSchema['fields']>;
+
+const schema: FormSchema = {
+    fields: (): FormSchemaFields => ({
+        title: [requiredCondition],
+        startDate: [requiredCondition],
+        endDate: [requiredCondition],
+    }),
+    validation: (value) => {
+        if (isDefined(value?.startDate)) {
+            if ((compareDate(value?.startDate, value?.endDate) > 0)) {
+                return (_ts('analysis.cloneModal', 'endDateGreaterThanStartDate'));
+            }
+        }
+        return (undefined);
+    },
+};
+
+interface CloneProperties {
+    title: string;
+    endDate: string;
+    startDate: string;
+}
+
+interface Props {
+    onClose: () => void;
+    projectId: number;
+    analysisId: number;
+    analysisToClone: {
+        title: string;
+        startDate?: string;
+        endDate: string;
+    },
+    updateAnalysisList: () => void;
+}
+
+function AnalysisCloneModal(props: Props) {
+    const {
+        onClose,
+        projectId,
+        analysisId,
+        analysisToClone,
+        updateAnalysisList,
+    } = props;
+
+    const formValue: PartialForm<FormType> = useMemo(() => ({
+        title: analysisToClone.title,
+        startDate: analysisToClone.startDate,
+        endDate: analysisToClone.endDate,
+    }), [analysisToClone]);
+
+    const {
+        pristine,
+        value,
+        error,
+        onValueChange,
+        validate,
+        onErrorSet,
+    } = useForm(formValue, schema);
+
+    const {
+        pending: pendingAnalysisClone,
+        trigger: triggerAnalysisClone,
+    } = useLazyRequest<CloneProperties, unknown>({
+        url: `server://projects/${projectId}/analysis/${analysisId}/clone/`,
+        method: 'POST',
+        body: ctx => ctx,
+        onSuccess: () => {
+            onClose();
+        },
+        failureHeader: _ts('analysis.cloneModal', 'anaylsisCloneFailed'),
+    });
+
+    const handleSubmitButtonClick = useCallback(() => {
+        const { errored, error: err, value: val } = validate();
+        onErrorSet(err);
+        if (!errored && isDefined(val)) {
+            triggerAnalysisClone(val as FormType);
+        }
+        updateAnalysisList();
+    }, [triggerAnalysisClone, onErrorSet, validate, updateAnalysisList]);
+
+    return (
+        <Modal
+            onCloseButtonClick={onClose}
+            heading={_ts('analysis.cloneModal', 'cloneModalHeading')}
+            bodyClassName={styles.modal}
+            footerActions={(
+                <Button
+                    name={undefined}
+                    variant="primary"
+                    onClick={handleSubmitButtonClick}
+                    disabled={pristine || pendingAnalysisClone}
+                >
+                    {_ts('analysis.cloneModal', 'cloneButtonLabel')}
+                </Button>
+            )}
+        >
+            <TextInput
```

Comment:
We should show NonFieldError.
We should also show loading animation.
We should also disable form when getting data or setting data.